src/mscorlib/shared/System/Guid.cs

```diff
@@ -1345,73 +1339,66 @@ public bool TryFormat(Span<char> destination, out int charsWritten, string forma
+                        *p++ = '0';
+                        *p++ = 'x';
+                        p += HexsToChars(p, _a >> 24, _a >> 16);
+                        p += HexsToChars(p, _a >> 8, _a);
+                        *p++ = ',';
+                        *p++ = '0';
+                        *p++ = 'x';
+                        p += HexsToChars(p, _b >> 8, _b);
+                        *p++ = ',';
+                        *p++ = '0';
+                        *p++ = 'x';
+                        p += HexsToChars(p, _c >> 8, _c);
+                        *p++ = ',';
+                        *p++ = '{';
+                        p += HexsToCharsHexOutput(p, _d, _e);
+                        *p++ = ',';
+                        p += HexsToCharsHexOutput(p, _f, _g);
+                        *p++ = ',';
+                        p += HexsToCharsHexOutput(p, _h, _i);
+                        *p++ = ',';
+                        p += HexsToCharsHexOutput(p, _j, _k);
+                        *p++ = '}';
                     }
                     else
                     {
                         // [{|(]dddddddd[-]dddd[-]dddd[-]dddd[-]dddddddddddd[}|)]
-                        offset += HexsToChars(guidChars + offset, _a >> 24, _a >> 16);
-                        offset += HexsToChars(guidChars + offset, _a >> 8, _a);
+                        p += HexsToChars(p, _a >> 24, _a >> 16);
+                        p += HexsToChars(p, _a >> 8, _a);
                         if (dash)
-                            guidChars[offset++] = '-';
-                        offset += HexsToChars(guidChars + offset, _b >> 8, _b);
+                            *p++ = '-';
+                        p += HexsToChars(p, _b >> 8, _b);
                         if (dash)
-                            guidChars[offset++] = '-';
-                        offset += HexsToChars(guidChars + offset, _c >> 8, _c);
+                            *p++ = '-';
+                        p += HexsToChars(p, _c >> 8, _c);
                         if (dash)
-                            guidChars[offset++] = '-';
-                        offset += HexsToChars(guidChars + offset, _d, _e);
+                            *p++ = '-';
+                        p += HexsToChars(p, _d, _e);
                         if (dash)
-                            guidChars[offset++] = '-';
-                        offset += HexsToChars(guidChars + offset, _f, _g);
-                        offset += HexsToChars(guidChars + offset, _h, _i);
-                        offset += HexsToChars(guidChars + offset, _j, _k);
+                            *p++ = '-';
+                        p += HexsToChars(p, _f, _g);
+                        p += HexsToChars(p, _h, _i);
+                        p += HexsToChars(p, _j, _k);
                     }
+
+                    if (braces != 0)
+                        *p++ = (char)(braces >> 16);
+
+                    Debug.Assert(p - guidChars == guidSize);
```

Comment:
I have refactored the code to make it possible to assert that we have written as many characters as expected. This refactoring also made the code smaller and a bit faster.